src/kernels/kernelFinder.node.ts

```diff
@@ -42,12 +42,6 @@ export class KernelFinder extends BaseKernelFinder {
 
     protected async isValidCachedKernel(kernel: KernelConnectionMetadata): Promise<boolean> {
         switch (kernel.kind) {
-            case 'startUsingRemoteKernelSpec':
-            case 'connectToLiveRemoteKernel':
-                // If this is a a remote kernel, it's valid if the URI is still active
-                const uri = await this.serverUriStorage.getRemoteUri();
```

Comment:
Not entirely sure why this was done, i guess its a temporary solution for #9167, but this breaks the code for AML (as AML uris don't work like this) also its very slow when loading the cache (making the cache completely useless).

This also slows down loading of local kernel specs as we load local & remote kernels from cache together, hence nohting appears until we validate the remote kernels, which feels like a poor design right now, but we can look into that in debt week.

Reply:
Without this the cache can be completely invalid. I suppose you're handling this by always returning the cache and letting the user pick the invalid items, but then throwing an error then? Basically move this validation to later in the process.

Reply:
Yes, thats correct.

Reply:
And not sure why the original code would have been slow. It's not actually going to the remote host, it's just querying the encrypted storage.

Reply:
Enable azure AML compute and the cache ends up setting up the azure compute connection, thats very slow and now none of the locally cached kernels load either for a long time.

Reply:
Also with azure AML compute this logic was incorrect, as the uri is not the remote jupyter uri, its something else.